controllers/profiles/common/ensurer.go

```diff
@@ -62,10 +62,10 @@ type defaultObjectEnsurer struct {
 	creator ObjectCreator
 }
 
-func (d *defaultObjectEnsurer) Ensure(ctx context.Context, workflow *operatorapi.SonataFlow, visitors ...MutateVisitor) (client.Object, controllerutil.OperationResult, error) {
+func (d *defaultObjectEnsurer) Ensure(ctx context.Context, workflow *operatorapi.SonataFlow, platform *operatorapi.SonataFlowPlatform, visitors ...MutateVisitor) (client.Object, controllerutil.OperationResult, error) {
 	result := controllerutil.OperationResultNone
 
-	object, err := d.creator(workflow)
+	object, err := d.creator(workflow, platform)
```

Comment:
Here `defaultObjectEnsurer` can hold the `platform` and do a nil check to call the creator in the same fashion:

```go
if (d.platform == nil) {
    object, err := d.creator(workflow)
} else {
   object, err := d.withPlatform(platform).creator(workflow)
}
```

This way I think we can preserve the API since mutators and creators having a new unused parameter might be hard to maintain as you said in the chat.

Reply:
I investigated this option and ATM I think it's not doable following your proposal, because the `creator` field is a function of type:
```go
type ObjectCreator func(workflow *operatorapi.SonataFlow) (client.Object, error)
```
and we cannot pass the `platform` reference to it, the signature is fixed.

What we can do, if you approve it, is to adopt an `interface` type instead, like:
```go
type ObjectCreator interface {
	Create(workflow *operatorapi.SonataFlow) (client.Object, error)
	WithPlatform(platform *operatorapi.SonataFlowPlatform) ObjectCreator
}
```
and then use it as:
```go
if (d.platform == nil) {
    object, err := d.creator.Create(workflow)
} else {
   object, err := d.creator.WithPlatform(platform).Create(workflow)
}
```

Of course, all the existing functions of type `ObjectCreator` have to be converted to `struct` with some basic implementation to hold the `platform` reference.

Reply:
Tried to accomodate the proposal in the latest commits, with few changes to properly pass-in the platform reference when it's needed. Working on the functional fixes now.